src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13010,7 +13010,7 @@ void Unit::ModSpellDurationTime(SpellInfo const* spellInfo, int32 & duration, Sp
 
     // called from caster
     if (Player* modOwner = GetSpellModOwner())
-        modOwner->ApplySpellMod(spellInfo->Id, SPELLMOD_DURATION, duration, spell);
+        modOwner->ApplySpellMod(spellInfo->Id, SPELLMOD_CASTING_TIME, duration, spell);
```

Comment:
at https://github.com/ShinDarth/TrinityCore/blob/fix/src/server/game/Entities/Unit/Unit.cpp this file looks like it ends (visual EOF) at the same point as this line change cuts here in the diff.